### PR TITLE
[mini] Implement delegate tramp info using gsharing context.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/mini/patch-info.h [hunk 1]
```diff
@@ -65,3 +65,4 @@ PATCH_INFO(SET_TLS_TRAMP, "set_tls_tramp")
 PATCH_INFO(JIT_ICALL_ADDR_NOCALL, "jit_icall_addr_nocall")
 PATCH_INFO(PROFILER_ALLOCATION_COUNT, "profiler_allocation_count")
 PATCH_INFO(PROFILER_CLAUSE_COUNT, "profiler_clause_count")
+PATCH_INFO(DELE_TRAMP, "delegate_trampoline_info")
```
Comment (vargaz) [inline mono/mini/patch-info.h]: Missing newline. Also, it could be called DEL_TRAMP or DELEGATE_TRAMP, DELE is not used as an abbrev currently.

Comment (vargaz) [reply mono/mini/patch-info.h]: Also, the aot file format version should be bumped in aot-runtime.h.

Comment (kumpera) [reply mono/mini/patch-info.h]: I think I need to update AOT patch serialization for this.

Comment (vargaz) [reply mono/mini/patch-info.h]: Yes

**Pull request merged**